### PR TITLE
feat(core): support block links on Bi-Directional Links

### PR DESCRIPTION
**Pull request by fundon**
Clsoes [AF-1348](https://linear.app/affine-design/issue/AF-1348/修复-bi-directional-links-里面的链接地址)

* Links to the current document should be ignored on `Backlinks`
* Links to the current document should be ignored on `Outgoing links`

https://github.com/user-attachments/assets/dbc43cea-5aca-4c6f-886a-356e3a91c1f1

### CODE REVIEW

**Comment by fundon** [comment]
* **#8192** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8192?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8169** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8169?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx [hunk 1]
```diff
@@ -89,12 +89,22 @@ const usePatchSpecs = (page: Doc, shared: boolean, mode: DocMode) => {
       const pageId = data.pageId;
       if (!pageId) return <span />;
 
+      let params: URLSearchParams | undefined = undefined;
+
+      if (data.params) {
+        const { mode, blockIds, elementIds } = data.params;
+        params = new URLSearchParams();
```
Comment (EYHN) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/lit-adaper.tsx]: can we simplify this to `new URLSearchParams(data.params);` ?

File: packages/frontend/core/src/modules/docs-search/schema.ts [hunk 2]
```diff
@@ -11,8 +11,11 @@ export const blockIndexSchema = defineSchema({
   blockId: 'String',
   content: 'FullText',
   flavour: 'String',
-  ref: 'String',
   blob: 'String',
+  // reference info
+  refDocId: 'String',
+  // `mode={}&blockIds={}&elementIds={}`
+  refParams: 'String',
```
Comment (EYHN) [inline packages/frontend/core/src/modules/docs-search/schema.ts:18]: split this to

refMode
refBlockIds
refElementIds

File: packages/frontend/core/src/modules/docs-search/schema.ts [hunk 3]
```diff
@@ -11,8 +11,11 @@ export const blockIndexSchema = defineSchema({
   blockId: 'String',
   content: 'FullText',
   flavour: 'String',
-  ref: 'String',
```
Comment (EYHN) [inline packages/frontend/core/src/modules/docs-search/schema.ts:14]: ref: `{"docId":"123","mode": ...}`

File: packages/frontend/core/src/modules/docs-search/worker/in-worker.ts [hunk 4]
```diff
@@ -130,28 +130,41 @@ async function crawlingDocData({
...
+          return null;
+        });
 
-        blockDocuments.push(
-          Document.from<BlockIndexSchema>(`${docId}:${blockId}`, {
-            docId,
-            flavour,
-            blockId,
-            content: text.toString(),
-            ref,
```
Comment (EYHN) [inline packages/frontend/core/src/modules/docs-search/worker/in-worker.ts:152]: refDocId: [xxx, xxx]

File: packages/frontend/core/src/modules/docs-search/schema.ts (hunk 3 above)
Comment (fundon) [reply packages/frontend/core/src/modules/docs-search/schema.ts:14]: keep `refDocId` for searching

**Review by EYHN — APPROVED** [review]

**Pull request merged**